src/Service/ProgrammesAggregationService.php

```diff
@@ -30,6 +30,11 @@ public function __construct(
         parent::__construct($repository, $mapper, $cache);
     }
 
+    public function countUpcomingStreamableDescendantEpisodes(Programme $programme): int
```

Comment:
I don't actually need the count in frontend, I want to see if any exist.
Should I also allow a limit to be passed in so that the query stops once it has found the first instance?
Or possible have a bool called `$exists` that limits the query to 1?

Reply:
In terms of perfomance should be fast to use a LIMIT in a query but you can't use it inside a COUNT, should be a normal SELECT, in this case you may want to rename the function and return a boolean.

Reply:
Why can't you use LIMIT in a COUNT?

Reply:
@votemike this is a good explanation https://stackoverflow.com/questions/17020842/mysql-count-with-limit

Reply:
I always thought you could use LIMIT with a COUNT.... Oops.

Reply:
So 3 options:
1.  SELECT with a LIMIT
2. Keep it as a COUNT
3. Use EXISTS